### DISCUSSION TITLE
Sending notifications from kittens

### DISCUSSION THREAD

**Discussion by eugene-babichenko** [kovidgoyal/kitty#5785] [Q&A]
What is the correct setup for sending a notification from a kitten?

I tried this:

```python
from typing import List
from kitty.boss import Boss
from kitty.notify import notify_with_command, NotificationCommand
from kittens.tui_handler import result_handler

def main(args: List[str]) -> str:
    pass

@result_handler(no_ui=True)
def handle_result(args: List[str], answer: str, target_window_id: str, boss: Boss) -> None:
    command = NotificationCommand(title= hello", body="world")
    notify_with_command(command, target_window_id)
```

but without any success.

**Comment by kovidgoyal** [comment]
See send_test_notification in boss.py

**Comment by eugene-babichenko** [comment]
I tried the code from the example

```python
from typing import List
from kittens.tui.handler import result_handler
from time import monotonic
from kitty.boss import Boss
from kitty.notify import notify

def main(args: List[str]) -> str:
    pass

@result_handler(no_ui=False)
def handle_result(args: List[str], answer: str, target_window_id: int, boss: Boss) -> None:
    now = monotonic()
    ident = f'test-notify-{now}'
    notify(f'Test {now}', f'At: {now}', identifier=ident, subtitle=f'Test subtitle {now}')
```

But it did not help. Running with `kitty +kitten $PWD/test.py`.

**Comment by kovidgoyal** [comment]
On Sat, Dec 24, 2022 at 11:26:00AM -0800, Yevhenii Babichenko wrote:
> I tried the code from the example
> 
> ```python
> from typing import List
> from kittens.tui.handler import result_handler
> from time import monotonic
> from kitty.boss import Boss
> from kitty.notify import notify
> 
> def main(args: List[str]) -> str:
>     pass
> 
> @result_handler(no_ui=False)
> def handle_result(args: List[str], answer: str, target_window_id: int, boss: Boss) -> None:
>     now = monotonic()
>     ident = f'test-notify-{now}'
>     notify(f'Test {now}', f'At: {now}', identifier=ident, subtitle=f'Test subtitle {now}')
> ```
> 
> But it did not help. Running with `kitty +kitten $PWD/test.py`.

This is a kitten that needs to run in kitty, via a mapping.

**Comment by page-down** [comment]
> ... sending a notification ...

You probably don't need to write a kitten.

https://sw.kovidgoyal.net/kitty/desktop-notifications/

> ... Running with kitty +kitten ...

If you need to run a `no_ui` kitten from the command line, use remote control.
This way kitten performs actions on a running kitty instance, rather than as a standalone CLI tool.

https://sw.kovidgoyal.net/kitty/remote-control/#kitty-kitten
